api/router/pubsub/ApplicationStatusHandler.go

```diff
@@ -100,7 +102,36 @@ func (impl *ApplicationStatusHandlerImpl) Subscribe() error {
 		if applicationDetail.StatusTime.IsZero() {
 			applicationDetail.StatusTime = time.Now()
 		}
-		isSucceeded, err := impl.appService.UpdateDeploymentStatusAndCheckIsSucceeded(app, applicationDetail.StatusTime)
+		isAppStoreApplication := false
+		_, err = impl.pipelineRepository.GetArgoPipelineByArgoAppName(app.ObjectMeta.Name)
+		if err != nil && err == pg.ErrNoRows {
+			impl.logger.Infow("this app not found in pipeline table looking in installed_apps table", "appName", app.ObjectMeta.Name)
+			//if not found in pipeline table then search in installed_apps table
+			gitOpsDeployedAppNames, err := impl.installedAppRepository.GetAllGitOpsDeploymentAppName()
+			if err != nil && err == pg.ErrNoRows {
+				//no installed_apps found
+				impl.logger.Errorw("no installed apps found", "err", err)
+				return
+			} else if err != nil {
+				impl.logger.Errorw("error in getting all gitops deployment app names from installed_apps ", "err", err)
+				return
+			}
+			var devtronGitOpsAppName string
+			gitOpsRepoPrefix := impl.appService.GetGitOpsRepoPrefix()
+			if len(gitOpsRepoPrefix) > 0 {
+				devtronGitOpsAppName = fmt.Sprintf("%s-%s", gitOpsRepoPrefix, app.ObjectMeta.Name)
+			} else {
+				devtronGitOpsAppName = app.ObjectMeta.Name
+			}
+			if slices.Contains(gitOpsDeployedAppNames, devtronGitOpsAppName) {
+				//app found in installed_apps table hence setting flag to true
+				isAppStoreApplication = true
+			} else {
+				//app neither found in installed_apps nor in pipeline table hence returning
+				return
+			}
+		}
+		isSucceeded, err := impl.appService.UpdateDeploymentStatusAndCheckIsSucceeded(app, applicationDetail.StatusTime, isAppStoreApplication)
```

Comment:
will we ever reach this line if `isAppStoreApplication`  is false? If not why do we need to pass it to `UpdateDeploymentStatusAndCheckIsSucceeded `